pkg/settlement/pseudosettle/pseudosettle.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package pseudosettle
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ethersphere/bee/pkg/logging"
+	"github.com/ethersphere/bee/pkg/p2p"
+	"github.com/ethersphere/bee/pkg/p2p/protobuf"
+	"github.com/ethersphere/bee/pkg/settlement"
+	pb "github.com/ethersphere/bee/pkg/settlement/pseudosettle/pb"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+const (
+	protocolName    = "pseudosettle"
+	protocolVersion = "1.0.0"
+	streamName      = "pseudosettle"
+)
+
+type Service struct {
+	streamer p2p.Streamer
+	logger   logging.Logger
+	observer settlement.PaymentObserver
+}
+
+type Options struct {
+	Streamer p2p.Streamer
+	Logger   logging.Logger
+}
+
+func New(o Options) *Service {
+	return &Service{
+		streamer: o.Streamer,
+		logger:   o.Logger,
+	}
+}
+
+func (s *Service) Protocol() p2p.ProtocolSpec {
+	return p2p.ProtocolSpec{
+		Name:    protocolName,
+		Version: protocolVersion,
+		StreamSpecs: []p2p.StreamSpec{
+			{
+				Name:    streamName,
+				Handler: s.handler,
+			},
+		},
+	}
+}
+
+func (s *Service) handler(ctx context.Context, p p2p.Peer, stream p2p.Stream) (err error) {
+	r := protobuf.NewReader(stream)
+	defer func() {
+		if err != nil {
+			_ = stream.Reset()
+		} else {
+			_ = stream.FullClose()
+		}
+	}()
+	var req pb.Payment
+	if err := r.ReadMsg(&req); err != nil {
+		return fmt.Errorf("read request from peer %v: %w", p.Address, err)
+	}
+
+	s.logger.Tracef("received payment message from peer %v of %d", p.Address, req.Amount)
```

Comment:
Why is this `trace`?

I think this should be on the `info` or at least `debug` level for sure!

The reason is that receiving a payment is something relevant to the user :)

Reply:
`info` seems to be way too spammy. during high load with one peer payment might be quite frequent.